pathiomock.go

```diff
@@ -0,0 +1,59 @@
+package pathio
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"sync"
+)
+
+// MockClient mocks out an S3 bucket
+type MockClient struct {
```

Comment:
Err, derp, I didn't read the whole thing, so I take that back. Though I still wonder what advantage this has over the old approach which was to create a temporary file to write to - is there something I'm missing?